packages/http-signature-utils/src/utils/signatures.ts

```diff
@@ -1,10 +1,9 @@
-import { sign, KeyLike, BinaryLike } from 'crypto'
-import {
-  httpis as httpsig,
-  Algorithm,
-  RequestLike,
-  Signer
-} from 'http-message-signatures'
+import { type KeyLike } from 'crypto'
+import { httpbis, createSigner, Request } from 'http-message-signatures'
+
+export interface RequestLike extends Request {
+  body?: string
+}
```

Comment:
The `Request` type from `http-message-signatures` does not contain the `body` property in its latest version.

Reply:
I found this commit: https://github.com/dhensby/node-http-message-signatures/commit/d23025305a970df239703b668c18477daf5e666d

> "There is no need for the body prop as signatures are not concerned with the body of the request.
All data required for signing/verifying http-signatures are based in the request line and headers."

Does this mean `body` is unnecessary in general?

I did a small test of this by updating the first tests in `validation.test.ts`, and it seems to be the case, but double check that.

Reply:
I'll double check this behaviour and follow up with my findings in a response here.

Reply:
Performed some tests withing the Web Monetization extension:
1. it looks like the body is unnecessary
2. adding `content-digest`, `content-length` and `content-type` is unnecessary as well (???)

I was able to make authenticated requests with the following headers:

- Without the `Authorization` header:
```
Accept: application/json
Content-Digest: sha-512=:IuQyFymTZ4iOpIw8X3Xy873ZzY8Z9Qv1RG7ZisWfvGMLRGt4a+rsHS313udn1vzJHE9Y6W/U8jeKFh11NiWAJw==:
Content-Type: application/json
Signature: sig1=:Hp0vZAXViJOlsIivP3YdAalmE+wu/zPpRFImAxcorgtOYlEei4dghnFtJTuVQWPoRlw/oNJKim9bASjODC9oDA==:
Signature-Input: sig1=("@method" "@target-uri");alg="ed25519";keyid="4dc5df20-270e-4dc1-90f1-b26c5cff1c1a";created=1708072371
```

- With the `Authorization` header:
```
Accept: application/json
Authorization: GNAP <redacted>
Content-Digest: sha-512=:dGOTVuK0RoQ19We9JvN+ttaHKBuWMTrTFbF+PxZLtlliGgd+IaoxWsVj4THmyhBuMqIGZSiFiCdEhKBjZCpM7g==:
Content-Type: application/json
Signature: sig1=:I1oaHWnVhiPX07f9r3z9j193Ne2XNuMwHazNQ7o4OknyWbx9nDOivJL7M76j2zEWohYfGs1mavMoyj8LwPDdAg==:
Signature-Input: sig1=("@method" "@target-uri" "authorization");alg="ed25519";keyid="4dc5df20-270e-4dc1-90f1-b26c5cff1c1a";created=1708072371
```

Following the [HTTP Signatures section of the GNAP spec](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol-16#section-7.3.1), it says that:
```
When the message contains a request body, the covered components MUST also include the following:

"content-digest": The Content-Digest header as defined in [[I-D.ietf-httpbis-digest-headers](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-digest-headers-13)].
When the request message has a body, the signer MUST calculate this field value and include the 
field in the request. The verifier MUST validate this field value. REQUIRED when the message 
request contains a message body.
```

Looking at a [request example from the GNAP specification](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol-16#section-7.3.1-25), I noticed it includes all `content-*` components in the `Signature-Input` header. However, I managed to make the requests successfully without including these components.

Reply:
> I was able to make authenticated requests with the following headers:

This makes sense because there is no `@content-digest` in the signature-input so it wasn't taken into account during signature generation or verification

Reply:
> "There is no need for the body prop as signatures are not concerned with the body of the request.
All data required for signing/verifying http-signatures are based in the request line and headers."

The way I understand this is that the body itself is not required for signing but the content headers are